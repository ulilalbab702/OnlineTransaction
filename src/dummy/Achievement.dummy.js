export const Achievement = [
    {
        "Code": "FM-JBYPM",
        "Name": "FMC Jembayan PAMA",
        "Percent": 50,
        "BackLogOpen": 1,
        "BackLogClose": 1,
        "Priority1": 1,
        "Priority2": 0,
        "Priority3": 1,
        "Priority4": 0
    }
]