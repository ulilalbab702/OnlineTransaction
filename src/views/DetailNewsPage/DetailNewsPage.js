import React, { useEffect, useRef, useState } from "react";
import { Row, Col } from "reactstrap";
import "./DetailNewsPage.scss";
import { ImagePromo } from "assets/images";
import { IconShareWhite } from "assets/icons";
import moment from "moment";
import Viewer from 'react-viewer';
import { isMobile } from "react-device-detect";
import { FaEnvelope, FaFacebookF, FaLink, FaShareAlt, FaTwitter, FaWhatsapp } from "react-icons/fa";
import { EmailShareButton, FacebookShareButton, TwitterShareButton, WhatsappShareButton } from "react-share";
import Breadcrumb from "components/Breadcrumb/Breadcrumb";
import firebase from "../../firebase/firebase";
import parse from "html-react-parser";
import { MENU } from "constants/menu";

const DetailNewsPage = (props) => {
  const [value, setValue] = useState("");
  const [keyword, setKeyword] = useState("");
  const [title, setTitle] = useState("");
  const [index, setIndex] = useState(0);
  const [newsId, setId] = useState(props.match.params.newsId);
  const [imageUrl, setImageUrl] = useState("");
  const [viewImage, setViewImage] = useState(false);
  const [imageAttribute, setImageAttribute] = useState(false);
  const [page, setPage] = useState(1);
  const [newsList, setDataRelated] = useState([]);
  const [dataNews, setDataNews] = useState({
    title: "Aplikasi UT Connect telah hadir di IOS",
    date: "20 Januari 2021",
    like: 129,
    image: ImagePromo,
    description: `United Tractors, Jakarta: Pada tahun 2020, United Tractors (UT) kembali berinovasi dengan meluncurkan aplikasi bertajuk: “UT Connect”. UT Connect adalah sebuah aplikasi perawatan alat berat yang dapat diakses melalui smartphone sehingga dapat memudahkan pelanggan untuk berkomunikasi langsung dengan perusahaan, mulai dari monitor alat berat yang mereka miliki hingga kebutuhan pelanggan akan informasi terkini yang berkaitan dengan pelayanan serta layanan United Tractos. 
            Fitur- Fitur UT Connect Berangkat dari inisiatif meningkatkan kepuasaan pelanggan, United Tractors melalui aplikasi perawatan alat berat, UT Connect, menyediakan fitur yang memudahkan pelanggan untuk mendapatkan feedback atas pelayanannya, yakni Ticketing System. 
            Tidak hanya itu, fitur ini pun dapat digunakan oleh pelanggan untuk melakukan pemesanan mekanik. UT Connect juga menawarkan fitur lainnya seperti Working History dan keamanan login menggunakan PIN. 
            Aplikasi perawatan alat berat, UT Connect dapat diunduh melalui PlayStore untuk Android dan Apple Store untuk iOS. Namun, sebelum mengakses aplikasi UT Connect, pelanggan harus melakukan registrasi terlebih dahulu dengan dua pilihan cara yang sangat mudah. Cara Registrasi UT Connect Registrasi bisa dilakukan dengan cara mengisi kertas formulir pendaftaran dengan membubuhkan tandatangan tinta basah untuk diserahkan kepada team UT After Sales Consultant (ASC) atau kantor cabang UT terdekat. Setelah itu, kami akan membantu membuatkan akun Anda. Apabila Anda telah memiliki satu akun dan berminat menambah akun lain, Anda dapat langsung melakukan pendaftaran secara mandiri melalui aplikasi UT Connect dengan memilih tombol “Register” yang tertera pada aplikasi. Untuk langkah seterusnya, Anda dapat melengkapi data diri seperti Nama, Nomor Handphone dan Customer Code pada e-form terlampir. UT Connect diluncurkan untuk mempermudah penggunanya dalam melakukan pemantauan alat berat miliknya agar dapat memperoleh informasi dan membantu operasi bisnis pelanggan. Segera daftarkan diri Anda untuk merasakan pelayanan terbaik yang ditawarkan oleh United Tractors. Download UT Connect disini. `,
  });
  const newsUrl = window.location.href;
  const cleanUrl = newsUrl;

  const setAnalyticClevertap = (action, event, screen, product) => {
    if (props.user) {
      const { userName } = props.user;
      if (product !== null) {
        window.clevertapEventProduct(`${product[0] + "_LogedIn"}`, product[1]);
      } else {
        window.clevertapEvent(`${event + "_LogedIn"}`, props.user);
      }
      firebase.analytics().setUserId(userName);
      firebase.analytics().setUserProperties(userName, "username:" + userName + `||View: ${screen}`);
      firebase.analytics().setCurrentScreen(screen, screen);
      firebase.analytics().logEvent(`${event + "_LogedIn"}`);
      firebase.analytics().setUserProperties("username", userName);
      firebase.analytics().setUserProperties(action, `${event + "_LogedIn"}`);
    } else {
      if (product !== null) {
        window.clevertapEventProduct(`${product[0] + "_NotLogedIn"}`, product[1]);
      } else {
        window.clevertapEvent(`${event + "_NotLogedIn"}`, null);
      }
      firebase.analytics().setCurrentScreen(screen, screen);
      firebase.analytics().logEvent(`${event + "_NotLogedIn"}`);
      firebase.analytics().setUserProperties(action, `${event + "_NotLogedIn"}`);
    }
  };

  useEffect(() => {
    setAnalyticClevertap("view", "View_ListNews_Screen", "View_ListNews_Screen", null);
    window.scrollTo(0, 0);
  }, []);

  useEffect(() => {
    async function fetchData() {
      await props.fetchDetailNews(newsId);
    }
    fetchData();
  }, []);

  useEffect(() => {
    if (props.detailNewsData) {
      setImageUrl(props.detailNewsData.imageNewsUrl)
    }
  }, [props.detailNewsData])

  useEffect(() => {
    async function fetchData() {
      await props.fetchNewsList(page, keyword, title);
    }
    fetchData();
  }, []);

  const _handleShowImage = () => {
    setAnalyticClevertap("view", "Click_ShareNewsDetail", "View_NewsDetail_Screen", ["Click_ShareNewsDetail", { "News_Name": props.detailNewsData.title }]);
    setViewImage(true);
    setImageAttribute(false);
  };
  const _handleCloseImage = () => {
    setViewImage(false);
  };
  const copyToClipboard = (text) => {
    let textField = document.createElement('textarea')
    textField.innerText = text
    document.body.appendChild(textField)
    textField.select()
    document.execCommand('copy')
    textField.remove()
  }

  const onCopyImageButtonClick = () => {
    copyToClipboard(cleanUrl);
    setImageAttribute(true);
  }
  const share = () => {
    if (navigator.share) {
      navigator
        .share({
          url: cleanUrl,
        })
        .then(() => {

        })
        .catch(error => {

        });
    }
  }
  const createImageViewToolbar = (toolbars) => {
    if (isMobile) {
      return toolbars.concat([
        {
          key: "copy",
          render: <FaLink />,
          onClick: () => { onCopyImageButtonClick() }
        },
        {
          key: "share",
          render: <FaShareAlt />,
          onClick: () => { share() }
        }

      ]);
    } else if (!isMobile) {
      return toolbars.concat([
        {
          key: "facebook",
          render: <FacebookShareButton url={cleanUrl}>
            <FaFacebookF />
          </FacebookShareButton>,
        },
        {
          key: "twitter",
          render: <TwitterShareButton url={cleanUrl}>
            <FaTwitter />
          </TwitterShareButton>,
        },
        {
          key: "wa",
          render: <WhatsappShareButton url={cleanUrl}>
            <FaWhatsapp />
          </WhatsappShareButton>,
        },
        {
          key: "email",
          render: <EmailShareButton url={cleanUrl}>
            <FaEnvelope />
          </EmailShareButton>,
        },
        {
          key: "copy",
          render: <FaLink />,
          onClick: () => onCopyImageButtonClick()
        },
      ]);
    }
  }

  const _handleChangeNews = (item) => {
    setAnalyticClevertap("click", "Click_NewsDetail", "View_NewsDetail_Screen", ["Click_NewsDetail", { "News_Name": item.title }]);
    setImageUrl(item.imageThumbnailUrl);
    setId(item.newsId);
    props.fetchDetailNews(item.newsId);
  }

  const _renderOtherNews = () => {
    return (
      <div className="default-margin">
        <div className='gridListDetailPromo'>
          <p className='title-news-relate'>Berita Terkait</p>
          {props.newsListData.data && props.newsListData.data.map((item) => {
            return (
              <div
                className='cardListPromotion'
                style={{ cursor: 'pointer' }}
                onClick={() => _handleChangeNews(item)}
              >
                <Col>
                  <img className='imageListPromotion' src={item.imageThumbnailUrl} />
                  <p className='titleProduct'>{item.title}</p>
                  <p className='seeDetail' >Lihat detail</p>
                </Col>
              </div>
            )
          })}
        </div>
        <p
          className='seeOther'
          onClick={() => {
            setAnalyticClevertap("click", "Click_SeeMoreNews", "View_NewsDetail_Screen", null);
            props.push(MENU.LIST_NEWS);
          }}
        >
          Berita lainnya →
        </p>
      </div>
    );
  };

  const _renderBreadcrumb = () => {
    const breadcrums = [
      {
        'url': '/Landing',
        'name': 'Landing'
      },
      {
        'url': '/Landing/ListNews',
        'name': 'List berita'
      }
    ];
    return (
      <div style={{ marginBottom: '0.5rem' }}>
        <Breadcrumb
          linkBreadcrumb={breadcrums}
          typography={"Detail berita"}
        />
      </div>
    )
  };

  return (
    <div className='defaultMargin ' style={{ paddingTop: 100, display: 'flex', flexDirection: 'column' }}>
      {_renderBreadcrumb()}
      <Viewer
        visible={viewImage}
        onClose={() => _handleCloseImage()}
        onMaskClick={() => _handleCloseImage()}
        images={[{ src: imageUrl, alt: 'Copied to clipboard: ' + cleanUrl }]}
        noNavbar={true}
        attribute={imageAttribute}
        zoomSpeed={0.1}
        noImgDetails={true}
        rotatable={false}
        scalable={false}
        changeable={false}
        showTotal={false}
        customToolbar={toolbars => createImageViewToolbar(toolbars)}
      />
      <p className='title-promotion-list'>{props.detailNewsData.title}</p>
      <p className='title-promotion-date'>
        {moment
          .utc(props.detailNewsData.createdDate)
          .local()
          .format("DD MMMM YYYY")}
      </p>
      <div
        style={{
          display: "flex",
          flexDirection: "row",
          alignItems: "center",
          marginBottom: 20,
          marginRight: "auto",
          cursor: "pointer"
        }}
        onClick={() => _handleShowImage()}
      >
        <span className="icon-share">
          <img src={IconShareWhite} />
        </span>
        <p className='title-icon' style={{ fontSize: "1vw", letterSpacing: 1 }}>Bagikan</p>
      </div>
      <div className='container-flex'>
        <div>
          <img src={imageUrl} className='image-promotion-list' />
          <div className="title-promotion-description">{parse(props.detailNewsData.description ? props.detailNewsData.description : "")}</div>
        </div>
        {_renderOtherNews()}
      </div>
    </div>
  );
};

export default DetailNewsPage;
