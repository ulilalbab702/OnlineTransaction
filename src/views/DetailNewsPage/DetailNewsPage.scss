@import "../../themes/colors";
@import "../../themes/fonts";
@import "../../themes/size";

.item-text-seeMore {
  font-style: normal;
  font-weight: bold;
  font-size: vw(16);
  color: #0055a0;
  margin-bottom: 0;
  margin-top: 0;
  margin-left: vw(5);
}
.container-flex {
  display: flex;
  flex-direction: row;
}
.icon-share {
  height: 30px;
  width: 30px;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: rgba(0, 0, 0, 0.7);
  margin-top: 5px;
  margin-left: 10px;
}
.title-promotion-list {
  font-style: normal;
  font-weight: bold;
  font-size: 2.8vw;
  line-height: 40px;
  letter-spacing: 0.02em;
  color: #0055a0;
  margin-bottom: 5px;
}
.title-promotion-date {
  font-style: normal;
  font-weight: 500;
  font-size: vw(16);
  line-height: 19px;
  color: #1d1d1d;
  opacity: 0.5;
}
.image-promotion-list {
  max-width: 100%;
  max-height: 100%;
  transform: scale(1) !important;
  border-radius: 10px;
}
.image-promotion-relate-list {
  max-width: 100%;
  max-height: 100%;
  transform: scale(1) !important;
}
.default-margin {
  margin-left: vw(60);
  margin-right: vw(40);
}
.title-product-related {
  font-style: normal;
  font-weight: bold;
  font-size: vw(20);
  letter-spacing: 0.01em;
  color: #303030;
  margin-top: vw(20);
}
.container-product-related {
  width: vw(250);
  display: flex;
  align-items: center;
  flex-direction: column;
}
.title-promotion-description {
  font-style: normal;
  font-weight: 500;
  font-size: vw(20);
  line-height: 25px;
  text-align: justify;
  letter-spacing: 0.01em;
  color: #404040;
  border: 0.1px solid #ffffff;
  box-sizing: border-box;
}
.input-gray {
  background-color: #f2f2f2 !important;
  border-radius: 5px !important;
  height: vw(70);
  display: flex;
  flex: 1;
  margin-left: vw(20);
  margin-right: vw(20);
  padding-left: vw(20);
  padding-right: vw(20);
  align-items: center;
}
.send-card {
  background-color: #ffd500 !important;
  border-radius: 5px !important;
  height: vw(50);
  width: vw(130);
  display: flex;
  flex-direction: row;
  align-items: center;
  padding-left: vw(20);
  padding-right: vw(20);
}
.title-send {
  font-family: "SFUIDisplay-Bold";
  margin-bottom: 0;
  font-weight: 500;
  font-size: vw(16);
  line-height: 22px;
  display: flex;
  align-items: center;
  letter-spacing: 0.4px;
  color: #000000;
  margin-right: vw(20);
}
.item-text-comment {
  font-style: normal;
  font-weight: 600;
  font-size: vw(16);
  line-height: 150%;
  letter-spacing: 0.5px;
  color: #000000;
  opacity: 0.7;
}
.item-name-comment {
  font-style: normal;
  font-weight: bold;
  font-size: vw(20);
  line-height: 24px;
  color: #515151;
  margin-bottom: 0;
  margin-left: vw(10);
  text-align: justify;
}
.item-date-comment {
  font-style: normal;
  font-weight: normal;
  font-size: vw(16);
  line-height: 19px;
  color: #b4b4b4;
  margin-left: vw(10);
}
.title-komentar {
  font-style: normal;
  font-weight: 600;
  font-size: vw(22);
  line-height: 26px;
  color: #404040;
  margin-bottom: 0;
  margin-left: vw(20);
}
.title-icon {
  font-style: normal;
  font-weight: bold;
  font-size: vw(16);
  line-height: 19px;
  color: rgba(0, 0, 0, 0.75);
  opacity: 0.8;
  margin-bottom: 0;
  margin-left: vw(10);
  margin-right: vw(10);
}
.line-vertical-comment {
  opacity: 0.15;
  border: 0.7px solid #000000;
  margin-left: 20px;
  margin-right: 20px;
  height: vw(40);
}
.title-news-relate {
  font-family: "SFProText-Bold";
  font-style: normal;
  font-weight: bold;
  font-size: vw(24);
  line-height: 40px;
  letter-spacing: 0.02em;
  color: #0055a0;
  margin-bottom: 0;
}
.title-see {
  font-style: normal;
  font-weight: bold;
  font-size: vw(24);
  line-height: 24px;
  color: #5d9fd6;
  text-align: center;
  margin-top: vw(40);
  cursor: pointer;
}
// Laptop 15 inch
@media only screen and (min-width: 1280px) {
  .item-text-seeMore {
    font-style: normal;
    font-weight: bold;
    font-size: vw(16);
    color: #0055a0;
    margin-bottom: 0;
    margin-top: 0;
    margin-left: vw(5);
  }
  .container-flex {
    display: flex;
    flex-direction: row;
  }
  .title-promotion-list {
    font-style: normal;
    font-weight: bold;
    font-size: 2.8vw;
    line-height: 40px;
    letter-spacing: 0.02em;
    color: #0055a0;
    margin-bottom: 5px;
  }
  .title-promotion-date {
    font-style: normal;
    font-weight: 500;
    font-size: vw(16);
    line-height: 19px;
    color: #1d1d1d;
    opacity: 0.5;
  }
  .image-promotion-list {
    max-width: 100%;
    max-height: 100%;
    transform: scale(1) !important;
    border-radius: 10px;
  }
  .image-promotion-relate-list {
    max-width: 100%;
    max-height: 100%;
    transform: scale(1) !important;
  }
  .default-margin {
    margin-left: vw(60);
    margin-right: vw(40);
  }
  .title-product-related {
    font-style: normal;
    font-weight: bold;
    font-size: vw(20);
    letter-spacing: 0.01em;
    color: #303030;
    margin-top: vw(20);
  }
  .container-product-related {
    width: vw(250);
    display: flex;
    align-items: center;
    flex-direction: column;
  }
  .title-promotion-description {
    font-style: normal;
    font-weight: 500;
    font-size: vw(20);
    line-height: 25px;
    text-align: justify;
    letter-spacing: 0.01em;
    color: #404040;
    border: 0.1px solid #ffffff;
    box-sizing: border-box;
  }
  .input-gray {
    background-color: #f2f2f2 !important;
    border-radius: 5px !important;
    height: vw(70);
    display: flex;
    flex: 1;
    margin-left: vw(20);
    margin-right: vw(20);
    padding-left: vw(20);
    padding-right: vw(20);
    align-items: center;
  }
  .send-card {
    background-color: #ffd500 !important;
    border-radius: 5px !important;
    height: vw(50);
    width: vw(130);
    display: flex;
    flex-direction: row;
    align-items: center;
    padding-left: vw(20);
    padding-right: vw(20);
  }
  .title-send {
    font-family: "SFUIDisplay-Bold";
    margin-bottom: 0;
    font-weight: 500;
    font-size: vw(16);
    line-height: 22px;
    display: flex;
    align-items: center;
    letter-spacing: 0.4px;
    color: #000000;
    margin-right: vw(20);
  }
  .item-text-comment {
    font-style: normal;
    font-weight: 600;
    font-size: vw(16);
    line-height: 150%;
    letter-spacing: 0.5px;
    color: #000000;
    opacity: 0.7;
  }
  .item-name-comment {
    font-style: normal;
    font-weight: bold;
    font-size: vw(20);
    line-height: 24px;
    color: #515151;
    margin-bottom: 0;
    margin-left: vw(10);
    text-align: justify;
  }
  .item-date-comment {
    font-style: normal;
    font-weight: normal;
    font-size: vw(16);
    line-height: 19px;
    color: #b4b4b4;
    margin-left: vw(10);
  }
  .title-komentar {
    font-style: normal;
    font-weight: 600;
    font-size: vw(22);
    line-height: 26px;
    color: #404040;
    margin-bottom: 0;
    margin-left: vw(20);
  }
  .title-icon {
    font-style: normal;
    font-weight: bold;
    font-size: vw(16);
    line-height: 19px;
    color: rgba(0, 0, 0, 0.75);
    opacity: 0.8;
    margin-bottom: 0;
    margin-left: vw(10);
    margin-right: vw(10);
  }
  .line-vertical-comment {
    opacity: 0.15;
    border: 0.7px solid #000000;
    margin-left: 20px;
    margin-right: 20px;
    height: vw(40);
  }
  .title-see {
    font-style: normal;
    font-weight: bold;
    font-size: vw(24);
    line-height: 24px;
    color: #5d9fd6;
    text-align: center;
    margin-top: vw(40);
  }
}

//mobile version
@media (max-width: 599px) {
  .container-flex {
    display: flex;
    flex-direction: column;
  }
  .item-text-seeMore {
    font-style: normal;
    font-weight: bold;
    font-size: vw-mobile(12);
    color: #0055a0;
    margin-bottom: vw-mobile(10);
    margin-top: 0;
    margin-left: vw-mobile(5);
  }
  .title-promotion-list {
    font-style: normal;
    font-weight: bold;
    font-size: vw-mobile(20);
    line-height: 40px;
    letter-spacing: 0.02em;
    color: #0055a0;
    margin-bottom: 0;
  }
  .title-promotion-date {
    font-style: normal;
    font-weight: 500;
    font-size: vw-mobile(14);
    line-height: 19px;
    color: #1d1d1d;
    opacity: 0.5;
  }
  .image-promotion-list {
    max-width: 100%;
    max-height: 100%;
    transform: scale(1) !important;
    border-radius: 10px;
  }
  .image-promotion-relate-list {
    max-width: 100%;
    max-height: 100%;
    transform: scale(1) !important;
  }
  .default-margin {
    margin-left: vw-mobile(20);
    margin-right: vw-mobile(20);
  }
  .title-product-related {
    font-style: normal;
    font-weight: bold;
    font-size: vw-mobile(16);
    letter-spacing: 0.01em;
    color: #303030;
    margin-top: vw-mobile(20);
  }
  .container-product-related {
    width: vw-mobile(250);
    display: flex;
    align-items: center;
    flex-direction: column;
  }
  .title-promotion-description {
    font-style: normal;
    font-weight: 500;
    font-size: vw-mobile(14);
    line-height: 25px;
    text-align: justify;
    letter-spacing: 0.01em;
    color: #404040;
    border: 0.1px solid #ffffff;
    box-sizing: border-box;
  }
  .input-gray {
    background-color: #f2f2f2 !important;
    border-radius: 5px !important;
    height: vw-mobile(50);
    display: flex;
    flex: 1;
    margin-left: vw-mobile(20);
    margin-right: vw-mobile(20);
    padding-left: vw-mobile(5);
    padding-right: vw-mobile(10);
    align-items: center;
    font-size: vw-mobile(10);
  }
  .send-card {
    background-color: #ffd500 !important;
    border-radius: 5px !important;
    height: vw-mobile(30);
    width: vw-mobile(90);
    display: flex;
    flex-direction: row;
    align-items: center;
    padding-left: vw-mobile(20);
    padding-right: vw-mobile(20);
  }
  .title-send {
    font-family: "SFUIDisplay-Bold";
    margin-bottom: 0;
    font-weight: 500;
    font-size: vw-mobile(12);
    line-height: 22px;
    display: flex;
    align-items: center;
    letter-spacing: 0.4px;
    color: #000000;
    margin-right: vw-mobile(20);
  }
  .item-text-comment {
    font-style: normal;
    font-weight: 600;
    font-size: vw-mobile(13);
    // letter-spacing: 0.5px;
    color: #000000;
    opacity: 0.7;
    text-align: justify;
    margin-bottom: vw-mobile(30);
  }
  .item-name-comment {
    font-style: normal;
    font-weight: bold;
    font-size: vw-mobile(16);
    line-height: 24px;
    color: #515151;
    margin-bottom: 0;
    text-align: justify;
  }
  .item-date-comment {
    font-style: normal;
    font-weight: normal;
    font-size: vw-mobile(12);
    line-height: 19px;
    color: #b4b4b4;
  }
  .title-komentar {
    font-style: normal;
    font-weight: 600;
    font-size: vw-mobile(16);
    line-height: 26px;
    color: #404040;
    margin-bottom: 0;
    margin-left: vw-mobile(10);
  }
  .title-icon {
    font-style: normal;
    font-weight: bold;
    font-size: vw-mobile(16);
    line-height: 19px;
    color: rgba(0, 0, 0, 0.75);
    opacity: 0.8;
    margin-bottom: 0;
    margin-left: vw-mobile(10);
    margin-right: vw-mobile(10);
  }
  .line-vertical-comment {
    opacity: 0.15;
    border: 0.7px solid #000000;
    margin-left: 20px;
    margin-right: 20px;
    height: vw-mobile(40);
  }
  .title-see {
    font-style: normal;
    font-weight: bold;
    font-size: vw-mobile(18);
    line-height: 24px;
    color: #5d9fd6;
    text-align: center;
    margin-top: vw-mobile(20);
  }
}

//tab version
@media (min-width: 600px) and (max-width: 1025px) {
  .item-text-seeMore {
    font-style: normal;
    font-weight: bold;
    font-size: vw-tab(12);
    color: #0055a0;
    margin-bottom: 0;
    margin-top: 0;
    margin-left: vw-tab(5);
  }
  .container-flex {
    display: flex;
    flex-direction: column;
  }
  .title-promotion-list {
    font-style: normal;
    font-weight: bold;
    font-size: vw-tab(24);
    line-height: 40px;
    letter-spacing: 0.02em;
    color: #0055a0;
    margin-bottom: 0;
  }
  .title-promotion-date {
    font-style: normal;
    font-weight: 500;
    font-size: vw-tab(14);
    line-height: 19px;
    color: #1d1d1d;
    opacity: 0.5;
  }
  .image-promotion-list {
    max-width: 100%;
    max-height: 100%;
    transform: scale(1) !important;
    border-radius: 10px;
  }
  .image-promotion-relate-list {
    max-width: 100%;
    max-height: 100%;
    transform: scale(1) !important;
  }
  .default-margin {
    margin-left: vw-tab(170);
    margin-right: vw-tab(40);
  }
  .title-product-related {
    font-style: normal;
    font-weight: bold;
    font-size: vw-tab(20);
    letter-spacing: 0.01em;
    color: #303030;
    margin-top: vw-tab(20);
  }
  .container-product-related {
    width: vw-tab(250);
    display: flex;
    align-items: center;
    flex-direction: column;
  }
  .title-promotion-description {
    font-style: normal;
    font-weight: 500;
    font-size: vw-tab(16);
    line-height: 25px;
    text-align: justify;
    letter-spacing: 0.01em;
    color: #404040;
    border: 0.1px solid #ffffff;
    box-sizing: border-box;
  }
  .input-gray {
    background-color: #f2f2f2 !important;
    border-radius: 5px !important;
    height: vw-tab(40);
    display: flex;
    flex: 1;
    margin-left: vw-tab(20);
    margin-right: vw-tab(20);
    padding-left: vw-tab(20);
    padding-right: vw-tab(20);
    align-items: center;
  }
  .send-card {
    background-color: #ffd500 !important;
    border-radius: 5px !important;
    height: vw-tab(30);
    width: vw-tab(100);
    display: flex;
    flex-direction: row;
    align-items: center;
    padding-left: vw-tab(20);
    padding-right: vw-tab(20);
  }
  .title-send {
    font-family: "SFUIDisplay-Bold";
    margin-bottom: 0;
    font-weight: 500;
    font-size: vw-tab(12);
    line-height: 22px;
    display: flex;
    align-items: center;
    letter-spacing: 0.4px;
    color: #000000;
    margin-right: vw-tab(20);
  }
  .item-text-comment {
    font-style: normal;
    font-weight: 600;
    font-size: vw-tab(12);
    line-height: 150%;
    letter-spacing: 0.5px;
    color: #000000;
    opacity: 0.7;
  }
  .item-name-comment {
    font-style: normal;
    font-weight: bold;
    font-size: vw-tab(14);
    line-height: 24px;
    color: #515151;
    margin-bottom: 0;
    margin-left: vw-tab(10);
    text-align: justify;
  }
  .item-date-comment {
    font-style: normal;
    font-weight: normal;
    font-size: vw-tab(12);
    line-height: 19px;
    color: #b4b4b4;
    margin-left: vw-tab(10);
  }
  .title-komentar {
    font-style: normal;
    font-weight: 600;
    font-size: vw-tab(14);
    line-height: 26px;
    color: #404040;
    margin-bottom: 0;
    margin-left: vw-tab(20);
  }
  .title-icon {
    font-style: normal;
    font-weight: bold;
    font-size: vw-tab(16);
    line-height: 19px;
    color: rgba(0, 0, 0, 0.75);
    opacity: 0.8;
    margin-bottom: 0;
    margin-left: vw-tab(10);
    margin-right: vw-tab(10);
  }
  .line-vertical-comment {
    opacity: 0.15;
    border: 0.7px solid #000000;
    margin-left: 20px;
    margin-right: 20px;
    height: vw-tab(40);
  }
  .title-see {
    font-style: normal;
    font-weight: bold;
    font-size: vw-tab(24);
    line-height: 24px;
    color: #5d9fd6;
    text-align: center;
    margin-top: vw-tab(40);
  }
}
