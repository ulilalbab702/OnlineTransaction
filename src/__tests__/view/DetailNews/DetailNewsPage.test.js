import React from "react";
import toJson from "enzyme-to-json";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import DetailNewsPage from "views/DetailNewsPage/DetailNewsPage";
import { dataDetailNewsDummy, listDetailNewsDummy } from "./../../../dummy/DetailNews.dummy";

Enzyme.configure({ adapter: new Adapter() });
describe("Detail News Component", () => {
	const wrapper = shallow(
		<DetailNewsPage
			detailNewsData={dataDetailNewsDummy}
			newsListData={listDetailNewsDummy}

		/>
	);
	describe("Detail News Component", () => {
		it("Should match to snapshot", () => {
			expect(toJson(wrapper)).toMatchSnapshot();
		});
	});
});
