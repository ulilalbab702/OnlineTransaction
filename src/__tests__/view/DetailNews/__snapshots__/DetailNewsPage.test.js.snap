// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail News Component Detail News Component Should match to snapshot 1`] = `
<div
  className="defaultMargin"
>
  <p
    className="title-promotion-list"
  />
  <p
    className="title-promotion-date"
  >
    12 April 2021
  </p>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyItems": "center",
        "marginBottom": 20,
      }
    }
  >
    <img
      height="20"
      src="likes.png"
      width="20"
    />
    <p
      className="title-icon"
    >
      20
    </p>
    <div
      className="line-vertical-comment "
    />
    <img
      height="20"
      src="chat.png"
      width="20"
    />
    <p
      className="title-icon"
    >
      129
    </p>
    <div
      className="line-vertical-comment "
    />
    <img
      height="20"
      src="share.png"
      width="20"
    />
    <p
      className="title-icon"
    >
      Share
    </p>
  </div>
  <div
    className="container-flex"
  >
    <div>
      <img
        className="image-promotion-list"
      />
      <p
        className="title-promotion-description"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 20,
          }
        }
      >
        <img
          height="20"
          src="chat.png"
          width="20"
        />
        <p
          className="title-komentar"
        >
          Komentar
        </p>
      </div>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <WithStyles(ForwardRef(Paper))
          className="input-gray"
        >
          <WithStyles(ForwardRef(Input))
            className="input-gray"
            disableUnderline={true}
            onChange={[Function]}
            placeholder="Tambahkan Komentar..."
            value=""
          />
          <WithStyles(ForwardRef(Paper))
            className="send-card"
            onClick={[Function]}
          >
            <p
              className="title-send"
            >
              Send
            </p>
            <img
              alt="icon"
              height="15"
              src="arrowSend.png"
              width="15"
            />
          </WithStyles(ForwardRef(Paper))>
        </WithStyles(ForwardRef(Paper))>
      </Row>
      <div>
        <div
          className="default-margin-top-bottom"
          key="0"
        >
          <div
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <img
              height="50"
              src="account-pic.png"
              width="50"
            />
            <div>
              <p
                className="item-name-comment"
              >
                Setiadi
              </p>
              <p
                className="item-date-comment"
              >
                2/22/2021, 8:41:17 AM
              </p>
            </div>
          </div>
          <i
            className="item-text-comment"
          >
            Ini Promo apakah masih berlaku ?
          </i>
          <p
            className="line-horizontal "
          />
        </div>
      </div>
    </div>
    <div
      className="default-margin"
    >
      <p
        className="title-news-relate"
      >
        Berita Terkait
      </p>
      <div
        className="container-product-related"
        key="daaa19b6-d0a6-460c-12c1-08d8d4877ae9"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <img
          className="image-promotion-relate-list"
          src="https://firebasestorage.googleapis.com/v0/b/customer-portal-dan-ut-portal.appspot.com/o/Images%2F307949f7-1787-4ff9-97a4-410c6687d037%2F.jpg?alt=media&token=368a6249-238f-4c67-8352-a800e29c415a"
        />
        <p
          className="title-product-related"
        >
          Aplikasi Perawatan Alat Berat UT Connect, Mudahkan Akses Informasi Produk United Tractors
        </p>
      </div>
      <div
        className="container-product-related"
        key="cb678399-9d98-4b60-12c4-08d8d4877ae9"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <img
          className="image-promotion-relate-list"
          src="https://firebasestorage.googleapis.com/v0/b/customer-portal-dan-ut-portal.appspot.com/o/Images%2F307949f7-1787-4ff9-97a4-410c6687d037%2F.jpg?alt=media&token=368a6249-238f-4c67-8352-a800e29c415a"
        />
        <p
          className="title-product-related"
        >
          Aplikasi Perawatan Alat Berat UT Connect, Mudahkan Akses Informasi Produk United Tractors
        </p>
      </div>
      <div
        className="container-product-related"
        key="9871434b-46ff-417d-12c2-08d8d4877ae9"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <img
          className="image-promotion-relate-list"
          src="https://firebasestorage.googleapis.com/v0/b/customer-portal-dan-ut-portal.appspot.com/o/Images%2F307949f7-1787-4ff9-97a4-410c6687d037%2F.jpg?alt=media&token=368a6249-238f-4c67-8352-a800e29c415a"
        />
        <p
          className="title-product-related"
        >
          Aplikasi UT Connect Telah hadir di IOS
        </p>
      </div>
      <p
        className="title-see"
        onClick={[Function]}
      >
        See Other Promo â†’
      </p>
    </div>
  </div>
</div>
`;
